src/ansys/dpf/post/selection.py

```diff
@@ -0,0 +1,557 @@
+            )
+
+        op = operators.scoping.transpose(
+            mesh_scoping=scoping, meshed_region=mesh._meshed_region, inclusive=0
+        )
+        self._selection.add_operator(op)
+        self._selection.set_output_name(
+            _WfNames.scoping, op.outputs.mesh_scoping_as_scoping
+        )
+
+    def select_elements(self, elements: Union[List[int], Scoping]) -> None:
+        """Select elements using their IDs or an elemental mesh scoping.
+
+        Parameters
+        ----------
+        elements :
+            element IDs or elemental mesh scoping.
+        """
+        if isinstance(elements, Scoping):
+            scoping = elements
+        else:
+            scoping = Scoping(
+                location=locations.elemental, ids=elements, server=self._server
+            )
+        self.select_with_scoping(scoping)
+
+    def intersect(
+        self, spatial_selection: Union[Selection, SpatialSelection, Scoping]
+    ) -> None:
+        """Replaces the current selection by its intersection with the input.
+
+        Parameters
+        ----------
+        spatial_selection:
+            Spatial domain to intersect with.
+
+        """
+        if isinstance(spatial_selection, Selection):
+            spatial_selection = spatial_selection.spatial_selection
+        elif isinstance(spatial_selection, Scoping):
+            spatial_selection = SpatialSelection(
+                scoping=spatial_selection, server=self._server
+            )
+
+        intersect_op = operators.scoping.intersect(server=self._server)
+
+        new_wf = Workflow(self._server)
+        new_wf.add_operator(intersect_op)
+        new_wf.set_input_name(_WfNames.scoping_a, intersect_op.inputs.scopingA)
+        new_wf.set_input_name(_WfNames.scoping_b, intersect_op.inputs.scopingB)
+        new_wf.set_output_name(_WfNames.scoping, intersect_op.outputs.intersection)
+        new_wf.connect_with(self._selection, {_WfNames.scoping: _WfNames.scoping_a})
+        new_wf.connect_with(
+            spatial_selection._selection, {_WfNames.scoping: _WfNames.scoping_b}
+        )
+        self._selection = new_wf
+
+    def _evaluate_on(self, simulation: Simulation) -> Union[Scoping, None]:
+        """Performs the currently defined selection on the given Simulation.
+
+        This scoping is internally used to evaluate result on the right spatial domain.
+
+        Parameters
+        ----------
+        simulation:
+            PyDPF-Post Simulation to apply the selection on.
+
+        Returns
+        -------
+        Scoping:
+            Resulting time/freq scoping. Returns ``None`` if no selection was made beforehand.
+
+        """
+        if self._selection is None:
+            return None
+        input_names = self._selection.input_names
+        if (
+            simulation._model.metadata.streams_provider is not None
+            and _WfNames.streams in input_names
+        ):
```

Comment:
This code seems very similar to ``_evaluate_on`` of the time scoping. We should either inheritate everything from another class, or use an external function for this.